tests/panic_source_locations.rs

```diff
@@ -0,0 +1,39 @@
+#[allow(unused_imports)]
+use cradle::prelude::*;
+#[allow(unused_imports)]
+use pretty_assertions::assert_eq;
+use std::{
+    panic::{set_hook, take_hook},
+    sync::{Arc, Mutex},
+};
+
+#[rustversion::since(1.46)]
+#[test]
+fn panics_contain_source_locations_of_run_and_run_output_call() {
+    let f = || run!("false");
+    let panic_location = get_panic_location(f);
+    assert_eq!(
+        Some("tests/panic_source_locations.rs:13:16".to_string()),
+        panic_location
+    );
+    let f = || run_output!("false");
+    let panic_location = get_panic_location(f);
+    assert_eq!(
+        Some("tests/panic_source_locations.rs:19:16".to_string()),
+        panic_location
+    );
+}
+
+#[allow(dead_code)]
+fn get_panic_location(f: fn()) -> Option<String> {
```

Comment:
I think it'd make more sense if this function returned `Option<Location>` instead of `Option<String>`. But I couldn't make the lifetimes work. @casey: Are you up for a little lifetime puzzle? :) (I also think it's ok like this.)

Reply:
I think `String` is okay too, which is convenient, because I don't think it's possible to solve the lifetime issues.